src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -1516,4 +1518,23 @@ public void validateDeleteRequest(Dataset dataset, List<DataDeletionTableModel>
             }
         }
     }
+
+    private static final String rowCountTemplate = "SELECT COUNT(<rowId>) FROM `<project>.<dataset>.<table>`";
+
+    public Map<String, Long> getSnapshotTableRowCounts(Snapshot snapshot) throws InterruptedException {
+        BigQueryProject bigQueryProject = bigQueryProjectForSnapshot(snapshot);
+        Map<String, Long> rowCounts = new HashMap<>();
+        for (SnapshotTable snapshotTable : snapshot.getTables()) {
```

Comment:
You shouldn't need to do all of those queries. A single query on the PDAO_ROW_ID_TABLE would give you all of the ids in one pass. Like:
```
select PDAO_TABLE_ID_COLUMN, count(PDAO_ROW_ID_COLUMN)
from PDAO_ROW_ID_TABLE
group by PDAO_TABLE_ID_COLUMN
```
and put the output straight into your map. You'd have to make a corresponding change to use the table id instead of the name on the metadata side.

Reply:
I thought about this, definitely cleaner to do it in one go but it doesn't account for hard deleted rows. Since we need to do that soon I figured better to handle it now.

Reply:
OK, perhaps make a comment to make that clear?